### ISSUE TITLE
docs: copy-constructor vs copy-assignment

### ISSUE THREAD

**Issue by El-even-11** [cmu-db/15445-bootcamp#5]
https://github.com/cmu-db/15445-bootcamp/blob/e2d462b821e29440da1f038e61589866c32d9dad/src/shared_ptr.cpp#L79-L80

Might use copy-constructor here?

**Comment by El-even-11** [comment]
```cpp
// copy-constructed
std::shared_ptr<Point> s4 = s3;

// copy-assigned
std::shared_ptr<Point> s4;
s4 = s3;
```

**Comment by cscourage** [comment]
I totally agree with you too.
"std::shared_ptr<Point> s4 = s3; " should be copy-construction, because it is the first time s4 appears.

"std::shared_ptr<Point> s4;
s4 = s3;"
and this should be copy-assignment.

**Comment by skyzh** [comment]
yep, this looks like a typo. would you please submit a pull request? thanks!
